src/App.jsx

```diff
@@ -1,14 +1,71 @@
 import './App.css';
+import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
+import { useState } from 'react';
+
+// count like feature
+const calculateTotalLikes = (messages) =>
+  messages.reduce((total, message) => total + (message.liked ? 1 : 0), 0);
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [localColor, setLocalColor] = useState('color');
+  const [remoteColor, setRemoteColor] = useState('color');
+
+  // Determine local and remote participants dynamically
+  const localSender = messages.length > 0 ? messages[0].sender : 'Local';
+  const remoteSender = [...new Set(messages.map((msg) => msg.sender))]
+    .find((sender) => sender !== localSender) || 'Remote';
```

Comment:
Very nice, @TatyanaA90! This a great way to dynamically set the local and remote sender! I love how you used the `Set` to ensure that you don't look at the same name multiple times and the default values as well! An interesting challenge could be how would this work for a group chat with multiple, more than 2, users?